src/m365/spo/commands/tenant/tenant-site-membership-list.spec.ts

```diff
@@ -0,0 +1,386 @@
+  ];
+  const adminUrl = 'https://contoso-admin.sharepoint.com';
+  const siteUrl = 'https://contoso.sharepoint.com/sites/site';
+  const siteId = '00000000-0000-0000-0000-000000000010';
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').resolves();
+    sinon.stub(telemetry, 'trackEvent').returns();
+    sinon.stub(pid, 'getProcessName').returns('');
+    sinon.stub(session, 'getId').returns('');
+    sinon.stub(spo, 'getRequestDigest').resolves({
+      FormDigestValue: 'ABC',
+      FormDigestTimeoutSeconds: 1800,
+      FormDigestExpiresAt: new Date(),
+      WebFullUrl: 'https://contoso.sharepoint.com'
+    });
+    auth.connection.active = true;
+    auth.connection.spoUrl = 'https://contoso.sharepoint.com';
+    commandInfo = cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogSpy = sinon.spy(logger, 'log');
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.post
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.connection.active = false;
+    auth.connection.spoUrl = undefined;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.TENANT_SITE_MEMBERSHIP_LIST);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if the siteUrl option is not a valid SharePoint URL', () => {
+    const actual = command.validate({ options: { siteUrl: 'foo' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the role option is not a valid role', () => {
+    const actual = command.validate({ options: { siteUrl: 'https://contoso.sharepoint.com', role: 'foo' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('lists all site membership groups', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `${adminUrl}/_api/SPO.Tenant/sites/GetSiteUserGroups?siteId='${siteId}'&userGroupIds=[0,1,2]`) {
+        return { value: [{ userGroup: ownerMembershipList }, { userGroup: membersMembershipList }, { userGroup: visitorsMembershipList }] };
+      };
+
+      throw 'Invalid request';
+    });
+
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `${adminUrl}/_api/SPO.Tenant/GetSitePropertiesByUrl`) {
+        return `{ "SiteId": "${siteId}" }`;
+      };
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: siteUrl } });
+    assert(loggerLogSpy.calledWith({
+      AssociatedOwnerGroup: ownerMembershipList,
+      AssociatedMemberGroup: membersMembershipList,
+      AssociatedVisitorGroup: visitorsMembershipList
+    }));
```

Comment:
This works indeed, the annoying part here imo is, when the test fails, you have 0 idea why. You only know that the logger has been called with something else. Therefore I like another approach a bit more.

```
assert.deepStrictEqual(loggerLogSpy.lastcall.args[0], <The object you expect>)
```

When we do this, if the test fails, we get a clear difference between what we expect, and what was passed to the logger. Let's update this in other tests as well.

Reply:
ok, that is a better idea. I have changed it